### ISSUE TITLE
Azimuth block gets stuck on many ships following recent OTA

### ISSUE THREAD

**Issue by dr-frmr** [urbit/urbit#5748]
This is an issue that I personally experienced on my ship and have seen in several others at this point. The issue is diagnosed by running `+azimuth-block` in dojo. If it's significantly behind (mine is `14.719.179` at time of writing) and doesn't increase regularly, it may be stuck. 

The block being stuck means that you can't communicate with ships whose network keys have changed since the block you're stuck on. This sequence of commands seems to fix the problem.
```
|nuke %eth-watcher
|start %eth-watcher
:azimuth %resub
:azimuth|watch 'http://eth-mainnet.urbit.org:8545' %default
```
I breached recently and notice that there are several ships that I can no longer communicate with. I suspect they may be experiencing this issue without knowing -- not sure what the best way to disseminate this info is.

**Comment by tacryt-socryp** [comment]
Can confirm that this happened to me and that the fix did indeed fix it. This is a major issue. I don't know what part of the state machine is getting messed up, but it's an extremely fundamental issue. We must have properly working networking for any of Ames' constraints to work. @belisarius222

**Comment by belisarius222** [comment]
This seems like a problem with `%eth-watcher`.  Maybe its timer is getting stuck.

@dr-frmr How recent was your breach when this happened?

**Comment by vvisigoth** [comment]
I also was behind (couldn't connect with ~haddef-sigwen who had breached) and I ran these commands. I seem to still be stuck. Any ideas? I'm happy to jump on a call with someone to troubleshoot and collect data. 

Also getting a ton of this in the dojo.

```
sivc-de-punt
sivc-de-punt
sivc-de-punt
sivc-de-punt
sivc-de-punt
sivc-de-punt
sivc-de-punt
sivc-de-punt
sivc-de-punt
sivc-de-punt
```

**Comment by vvisigoth** [comment]
Update on this. 

Tried all above commands, checked my timers, set up a personal infura node, pointed eth-watcher at it and my `(lent logs)` remain stuck at 0. I also observe (on the infura dashboard) that no requests have actually hit infura from my ship. And I checked that the instance could hit infura using curl, so it's purely an Urbit problem.

What more can I try to debug this? There's no way to tell how many people are out there who can't communicate because of these log issues, so it's high priority to figure this out. @lukestiles @philipcmonk

**Comment by vvisigoth** [comment]
I should also point out that my ship bogs down, spinning on `<<http>>` and when I Ctrl-C out of it, I get an eth-watcher stack trace. So this is likely also causing performance problems.

**Comment by yosoyubik** [comment]
Based on what @vvisigoth has shared with me it looks like:

- the %base desk is running all the supposed agents—`%hodl` was never installed, so it doesn't look like any other agent took over ames.
- `/app/azimuth` and `/app/eth-watcher` are running, with `%azimuth` subscribed to `%eth-watcher`
- `%eth-watcher` is running new threads that seem to get restarted
- @vvisigoth has tried with the default Infura proxy (`http://eth-mainnet.urbit.org:8545`), using our Infura endpoint directly, and his own Infura node.

Some other things we could try :
- Confirm that the thread's config for the azimuth thread looks ok, and it didn't somehow ended up not using the right contracts: `:eth-watcher +dbug [%state '-:(~(got by dogs) /azimuth)']`
- Add some sigpams in `/ted/eth-watcher` to confirm that it's actually receiving some logs (although it looks like it's not doing any calls to the node at all)

**Comment by vvisigoth** [comment]
(btw we tried all that stuff and it looks fine)

A note, I've been running a local ship on a laptop. It has been running for cumulative several hours and its log length is still at 0. I booted it in Port but once I confirmed that the issues were not Port specific I ejected and am now running CLI.

I experienced the exact thing where I was able to join a group immediately and have been unable to connect since. I can't even `|hi ~zod` leading me to believe that this is a key propagation issue. 

Another oddity: running `+keys` on other ships reveals that they think that the troublesome ship is on life 2. The ship itself thinks it's on life 1. The troublesome ship also seems to have correct life and rift for other ships. So it's most likely trying to communicate on its old key; could it be that it somehow hasn't heard of its own breach?

In any case, this is a major issue. As it is, it can take 2 days for a hosted ship to get a breach notification and local ships are functionally incommunicado ... indefinitely. 

There is a PR that _might_ address this (78?). Are there other lines being explored? What is the ETA on that PR going out? @philipcmonk @yosoyubik @lukestiles

**Comment by Fang-** [comment]
> Another oddity: running +keys on other ships reveals that they think that the troublesome ship is on life 2. The ship itself thinks it's on life 1. The troublesome ship also seems to have correct life and rift for other ships. So it's most likely trying to communicate on its old key; could it be that it somehow hasn't heard of its own breach?

Either it hasn't heard its own key change, or it doesn't know what the post-change keys are.

Just to get an additional data-point here, if you run this scry, does it succeed or crash/fail? (Success will print your private key for life 2, obviously don't share this.)

```
.^(ring %j /=vein=/2)
```

If it succeeds, this means your ship knows the new keys, and will switch to them as soon as it learns it is on life 2.  
If it fails, this means jael doesn't know the keys for life 2, and so will stay on life 1 even when it learns you're actually on life 2.

**Comment by vvisigoth** [comment]
It succeeds. I guess the breacher is the last to know about the breach?

**Comment by yosoyubik** [comment]
It looks like the thread tried 10 times and failed—I think it should keep retrying, would be good to confirm that—but, what's causing those failures, Infura node being overloaded? Firewall issue on the ship? Server connectivity? (@Fang- something looking weird on our Infura node?)

> I've been running a local ship on a laptop. It has been running for cumulative several hours and its log length is still at 0

I had several times for the past weeks booted moons from scratch and they all get up to date with azimuth within ~1-2 hours (this is on digital ocean) and I have tried also with comets locally on my laptop so, what's different for that local ship's environment? I don't know how Infura's set up work but maybe we shouldn't use one proxy for all ships in the network?

**Comment by Fang-** [comment]
> (@Fang- something looking weird on our Infura node?)

No, nothing seems out of the ordinary afaict. And generally, if there is, the network more broadly is affected.

**Comment by vvisigoth** [comment]
```
I had several times for the past weeks booted moons from scratch and they all get up to date with azimuth within ~1-2 hours (this is on digital ocean)
```

I hope you're not saying that it effectively takes a ship even one hour to be usable. That's not going to fly.

Try some L2 planets. I have no idea how key propagation is different on all of these ship classes.

**Comment by philipcmonk** [comment]
Yes, it takes 1-2 hours for a new ship to get up to date, and it will often be spinning on `<<http>>` during that time.  If you ^C it, then it'll start over from the beginning in a few minutes.

This will be improved in the next %base release (ETA a couple weeks-ish) to around 15min, and that will also give us the tools to improve it more.

**Comment by thelifeandtimes** [comment]
I was stuck at `+azimuth-block` ~14.651.300 for about the last 5 weeks. Waited to see if the most recent ota would rectify this, but to no avail. So I tried the `|nuke` and resubbing which pushed my `+azimuth-block` back to the ~13.000.000 range and getting various ames and http spinners. It did this for about an hour and then seems like it ran out of memory:
```
~sarlev-sarsen:dojo> :azimuth|watch 'http://eth-mainnet.urbit.org:8545' %defaultaddress 0x5925ec320 out of loom!
loom: [0x200000000 : 0x280000000)
Assertion '0' failed in noun/events.c:193

bail: oops
bailing out
pier: serf unexpectedly shut down
```
I have rebooted and am stuck with an http spinner, so will wait that out to see what happens, but this is increasingly an issue the further out of date my azimuth state gets from the rest of the network--particularly since it seems to imply that none of the NYC L2 planets that were distributed would be able to communicate with ships experiencing this issue (and because people may not know they have this issue unless they are looking for it).

Will messages that are attempting to be sent to over the past 5 weeks eventually get to me?

**Comment by thelifeandtimes** [comment]
I am now getting a growing stack of dropped packets:
```
ames: packet dropped (1000 total)
ames: packet dropped (2000 total)
ames: packet dropped (3000 total)
```

**Comment by thelifeandtimes** [comment]
Just pulled my ship off my droplet and onto a local machine, `meld`ed,  and tried `:azimuth|kick` which got me: 
```
>>  %no-logs-in-azimuth-state
[%jael-listen {} [%.n p=%azimuth]]
ship: processing azimuth snapshot (77.674 points)
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
ames: missing peer-state, ignoring
> :azimuth|kick
>=
```
So I tried to re-run a `|nuke` and `|start` then resubbing to my own infura endpoint. It looks like at some point earlier today that it had made ~1800 requests to my infura endpoint earlier in the day.

**Comment by thelifeandtimes** [comment]
@joemfb it was suggested by @yosoyubik that you might be able to help with the memory stuff here? He was going to try building a custom snapshot as well but wondering what else I can attempt here.

**Comment by joemfb** [comment]
@thelifeandtimes is that hang and subsequent stack trace on `ctrl-c` reproducible? does it regularly end in `/lib/dice/hoon:<[99 33].[99 53]>`? I don't see anything unusual about that particular line of code, and I wouldn't expect it to run for a really long time.

/cc @philipcmonk

**Comment by thelifeandtimes** [comment]
I don't think so. I just rebooted and ran a `ctrl-c` interrupt and got this: https://pastebin.com/5kMwM079

Not sure if that is because of when / how quickly after getting the http spinner that I interrupted it? I will note that this http spinner seemed to occur more quickly than previous ones, not letting the ship get through the full `ames: czar` sequence.

**Comment by Haroldthe24th** [comment]
I have been having similar issues ever since 1.9, I tried the suggested fix several times now over a period of a week, it's been 2+ weeks in total.

**Comment by vvisigoth** [comment]
@lukestiles @philipcmonk can we get an update on this? I think it's safe to say that we've seen the effects of 1.9 on this problem by now (if any).

**Comment by joemfb** [comment]
@thelifeandtimes also, the crash in your update-2 post is running out of memory (not being caught as such due to a pre-existing bug). Given the preceding output, this could also be caused by astrolabe.

**Comment by thelifeandtimes** [comment]
I have uninstalled astrolabe; what should I do next to kickstart the next stage? I'm over a million blocks behind--will I be able to make it through all of those without crashing?

**Comment by joemfb** [comment]
@thelifeandtimes can you run `|mass` and pastebin the output? maybe before and after a `|meld`? the biggest risk is just running out of memory, and that will depend on the space you have available.

**Comment by thelifeandtimes** [comment]
Pre-meld: https://pastebin.com/PbWvEAgB

output during meld: 
```
~sarlev-sarsen:dojo> hash-cons arena:
  root: B/160
  atoms (3554444):
    refs: MB/114.057.740
    data: MB/336.290.884
    dict: MB/281.966.240
  total: MB/732.314.864
  cells (26842676):
    refs: MB/781.763.380
    dict: GB/1.932.625.360
  total: GB/2.714.388.740
total: GB/3.446.703.764

pier: meld complete
```

Post-meld: https://pastebin.com/H3dvfUSk

May be worth noting that I `meld`ed using the binaries maybe a week or two ago during the lifetime of this issue to no avail. droped me from maybe 1.3GB to 1.1ish.

**Comment by thelifeandtimes** [comment]
uhhhh, noticed a few `gall: %ballot dozing, got %fact` then got over 1500 lines of `gall: %astrolabe dozing, got %fact` and then about 300 lines of a bunch of different `ames: breach peer [~sarlev-sarsen ~fipsyp-lagbec]` which maybe is a good sign?

**Comment by thelifeandtimes** [comment]
literally no idea what I did or didn't do but saw a bunch of notifs slam through on EScape twenty minutes ago and just checked:
```
> +azimuth-block
15.004.717
```
Looks like we are back in business....

**Comment by dchem** [comment]
I am having the same problem as @thelifeandtimes and I am tracking exactly the same behavior from my ship.

Mine also got stuck at `13.958.654` block. No miracle resolution for me yet.

**Comment by joemfb** [comment]
@thelifeandtimes that's great! did you uninstall astrolabe around the same time as your comment (https://github.com/urbit/urbit/issues/5748#issuecomment-1162279019), or had you done so earlier?

@dchem do your crashes look like any of the specific ones posted above? how much memory are you using per `|mass` (especially the last two lines `free lists: ...` and `sweep: ...`)? are you able to run `|meld`? (NB, it uses lots of RAM)

**Comment by thelifeandtimes** [comment]
> @thelifeandtimes that's great! did you uninstall astrolabe around the same time as your comment (https://github.com/urbit/urbit/issues/5748#issuecomment-1162279019), or had you done so earlier?

I had uninstalled it maybe 24 hours prior? Monday AM, PST.

**Comment by famousj** [comment]
I have `+azimuth-block` stuck at `13.958.654`

I've tried the suggested
```
|nuke %eth-watcher
|start %eth-watcher
:azimuth %resub
:azimuth|watch 'http://eth-mainnet.urbit.org:8545' %default
```
 Which didn't do the trick. I also tried `:azimuth|kick`.  I got this:

```
>>  %no-logs-in-azimuth-state
[%jael-listen {} [%.n p=%azimuth]]
ship: processing azimuth snapshot (77.674 points)
ames: missing peer-state, ignoring
```
After the "ignoring" it seems to stop doing anything.

Any suggestions?

**Comment by yosoyubik** [comment]
> @famousj %no-logs-in-azimuth-state

this was caused by the previous `:azimuth|watch` (it wipes the state and starts from the last downloaded log) but it's probably that you did not have any logs at all, so your eth-watcher agent was hitting the stuck `%behn` timers issue. I've found this difficult to reproduce (so I can't confirm that it's cause https://github.com/urbit/urbit/issues/5747 but there's already a fix for [that on](https://github.com/urbit/urbit/pull/5857) so once it goes out it'll hopefully fix it—fwiw, several people that have reported that they eventually start downloading logs (either with the default Infura proxy or their own) but it usually takes a long time; they happy path where the timers don't get stuck can take from ~m30 to ~h1

**Comment by famousj** [comment]
Indeed, it's been quite a bit longer than `~h1`.  I'll keep checking.  Thanks!

**Comment by jrl-oci** [comment]
Update: I seem to be getting this:
```
eth-watcher failed; will retry
%retry-too-many
```
Any suggestions?

**Comment by famousj** [comment]
Update: I seem to be getting this:
```
eth-watcher failed; will retry
%retry-too-many
```
Any suggestions?

**Comment by patosullivan** [comment]
I'm seeing the same issue and it's preventing me from spinning up new moons for mainnet development.

**Comment by ryjm** [comment]
i did the same with another ship, also stuck on the same block:

![image](https://user-images.githubusercontent.com/2065299/176035371-07e5da86-99e3-46e6-849f-2940f95dca30.png)

**Comment by famousj** [comment]
Rumor has it this is actually a problem with `behn`, and a fix will be rolling out shortly.

**Comment by ryjm** [comment]
i'm running the new behn on ~losteg-binreb now - still at the same block but yeah looks like the timer is consistently firing now. i'll check back in an hour.

**Comment by yosoyubik** [comment]
@ryjm great! very happy about this :)

> eth-watcher failed; will retry

this is usually an issue with the Infura proxy, so it's hitting the endpoint but the request is being dropped—it eventually gets resolved but kicking it (on the proxy/Infura side) seems to also fix it.

**Comment by matildepark** [comment]
Stuck on the extremely important `13.958.654` block on two separate ships now. Trying to figure out where the New Behn is such that I might be able to actually deploy an app running on a now-breached ship.

**Comment by ant0n-anon** [comment]
okay...i'm collecting sivc-de-punts on ~winder-dapsym. i have a repair scheduled tomorrow but i'm worried now reading this thread it might be more than we were thinking...

**Comment by Aoxian** [comment]
I experience issues with a stuck Azimuth block. The comments here were helpful.  At first I was at block 14.XXX.XXX.  I tried to `:azumuth|kick` but it was still stuck.

I decided to try the
```
|nuke %eth-watcher
|start %eth-watcher
:azimuth %resub
:azimuth|watch 'http://eth-mainnet.urbit.org:8545' %default
```

After running and waiting an hour, my block dropped to 13.XXX.XXX and I kind of panicked.  So I tried to kick it again:
```
:azimuth|kick
```

The result was my block updating to 15.XXX.XXX (which is good at the time of the post).  Everything started flowing again.

**Comment by minderfolden** [comment]
I'm experiencing this currently. On 1.10.  Tried kicking azimuth and gotthis:

```
clay: read-at-aeon fail [desk=%base care=%a case=[%da p=~2022.9.28..01.34.12..1ce4] path=/gen/azimuth/kick/hoon]
/sys/vane/clay/hoon:<[1.168 18].[1.168 45]>
/sys/vane/clay/hoon:<[1.168 9].[1.172 11]>
[%file-not-found-mist /gen/azimuth/kick/hoon]
/sys/vane/clay/hoon:<[1.167 9].[1.172 11]>
/sys/vane/clay/hoon:<[1.166 9].[1.172 11]>
/sys/vane/clay/hoon:<[1.164 7].[1.183 9]>
/sys/vane/clay/hoon:<[1.163 7].[1.183 9]>
/sys/vane/clay/hoon:<[1.204 19].[1.204 38]>
/sys/vane/clay/hoon:<[1.204 18].[1.204 43]>
/sys/vane/clay/hoon:<[1.204 7].[1.239 17]>
/sys/vane/clay/hoon:<[1.203 7].[1.239 17]>
/sys/vane/clay/hoon:<[1.202 7].[1.239 17]>
/sys/vane/clay/hoon:<[610 7].[623 25]>
/sys/vane/clay/hoon:<[609 7].[623 25]>
/sys/vane/clay/hoon:<[607 7].[623 25]>
/sys/vane/clay/hoon:<[606 7].[623 25]>
/sys/vane/clay/hoon:<[1.195 9].[1.195 15]>
/sys/vane/clay/hoon:<[1.194 7].[1.198 23]>
/sys/vane/clay/hoon:<[1.193 7].[1.198 23]>
/sys/vane/clay/hoon:<[605 7].[623 25]>
/sys/vane/clay/hoon:<[604 7].[623 25]>
[%error-validating /gen/azimuth/kick/hoon]
/sys/vane/clay/hoon:<[603 7].[623 25]>
/sys/vane/clay/hoon:<[602 7].[623 25]>
/sys/vane/clay/hoon:<[903 26].[903 42]>
/sys/vane/clay/hoon:<[903 7].[912 24]>
/sys/vane/clay/hoon:<[902 7].[912 24]>
/sys/vane/clay/hoon:<[900 7].[912 24]>
/sys/vane/clay/hoon:<[899 7].[912 24]>
/sys/vane/clay/hoon:<[898 7].[912 24]>
/sys/vane/clay/hoon:<[1.195 9].[1.195 15]>
/sys/vane/clay/hoon:<[1.194 7].[1.198 23]>
/sys/vane/clay/hoon:<[1.193 7].[1.198 23]>
/sys/vane/clay/hoon:<[897 7].[912 24]>
/sys/vane/clay/hoon:<[896 7].[912 24]>
[%error-building /gen/azimuth/kick/hoon]
/sys/vane/clay/hoon:<[895 7].[912 24]>
/sys/vane/clay/hoon:<[893 7].[912 24]>
/sys/vane/clay/hoon:<[892 7].[912 24]>
/sys/vane/clay/hoon:<[916 7].[916 32]>
clay: %a build failed [%base 35 /gen/azimuth/kick/hoon]
/sys/vane/clay/hoon:<[4.182 15].[4.182 36]>
/sys/vane/clay/hoon:<[4.181 9].[4.198 11]>
/sys/vane/clay/hoon:<[4.180 9].[4.198 11]>
/sys/vane/clay/hoon:<[4.174 29].[4.174 33]>
dojo: %writ fail [i=%~.hand t=/gen/azimuth/kick]
> :azimuth|kick
/gen/azimuth/kick/hoon
%generator-build-fail
```

**Comment by joemfb** [comment]
@minderfolden `:azimuth|kick` has been removed -- the new command to fix this is `-azimuth-load`.

**Comment by minderfolden** [comment]
Ah, thank you!

**Comment by minweb-datsyt** [comment]
I'm experiencing this also with 1.15. I've tried:

```
|nuke %eth-watcher
|start %eth-watcher
:azimuth %resub
:azimuth|watch 'http://eth-mainnet.urbit.org:8545' %default
-azimuth-load
```

Still stuck on [a block from August](https://etherscan.io/block/15373409):

```
> +azimuth-block
15.373.409
```

**Comment by minweb-datsyt** [comment]
My ship is back up to date. What I did was:

```
|pack
|meld
````

Then I shut down my ship (CTRL+D) and launched with `--loom 32` (4GB). I issued these commands over again:

```
|nuke %eth-watcher
|start %eth-watcher
:azimuth %resub
:azimuth|watch 'http://eth-mainnet.urbit.org:8545' %default
-azimuth-load
```

Sometime during the night it caught up.

```
> +azimuth-block
16.247.240
```

**Comment by yosoyubik** [comment]
I finally have a reproduction of this issue—it's different from the one that people report here but could be related—on a ship that I have access to. It looks like %eth-watcher is downloading new logs, so +azimuth-block reports the latest block, but /app/azimuth is not receiving them.

I have put together a diagnostics desk (`|install ~dister-norsyr-torryn %diagnose`; to run it `-diagnose!azimuth`) that uses a thread to print info about the last downloaded and processed blocks. If anybody runs it, please share the output here.

My output where the last processed blocks is way older that the last downloaded:

```
> -diagnose!azimuth
last downloaded %azimuth block: #16.354.060 on ~2023.1.7..09.38.23
last processed %azimuth block: #16.133.035 on ~2022.12.7..13.10.47
installed eth-watcher? %.y
running eth-watcher? %.y
...
(wait 5s. to run the thread again)

```

**Comment by belisarius222** [comment]
> or the subscription crashed

Do you mean an agent crashed while handling a %watch move?

**Comment by yosoyubik** [comment]
No, we were already subscribed, so somehow %eth-watcher lost the the subscription from %azimuth, but this doesn't look like a %kick because from the point of view of /app/azimuth we were still subscribed (re-subscribing wouldn't work in this case, and we would get a "wire not unique"). They only way here is nuking %eth-watcher which does remove all subscriptions to it, and then we can start it again, subscribe, re-download the snapshot... etc.

This case that I encounter might be too specific to some problems the ship was having but it might point to this happening somewhere else for other people (azimuth -> eth-watcher, eth-watcher -> spider ...). For context, this ship (a moon) had its dojo in a broken state—it would not accept pressing enter—and I fixed it by injecting an event that unlinked %dojo but, by looking at when was the last processed azimuth block, %eth-watcher lost the subscription from %azimuth around the same time.

**Comment by zalberico** [comment]
I've been helping `~witbep-witbep` in UC Help and he has an azimuth block stuck about a million back. I had him run the tool and it reported this:
```
 -diagnose!azimuth
/sys/vane/gall/hoon:<[1.254 9].[1.254 37]>
/app/spider/hoon:<[164 5].[176 17]>
/app/spider/hoon:<[165 5].[176 17]>
/app/spider/hoon:<[167 5].[176 17]>
/app/spider/hoon:<[168 7].[175 9]>
/app/spider/hoon:<[170 27].[170 78]>
/app/spider/hoon:<[306 3].[306 49]>
/app/spider/hoon:<[317 3].[355 5]>
/app/spider/hoon:<[318 3].[355 5]>
/app/spider/hoon:<[322 3].[355 5]>
/app/spider/hoon:<[331 3].[355 5]>
/app/spider/hoon:<[333 3].[355 5]>
/app/spider/hoon:<[336 3].[355 5]>
/app/spider/hoon:<[340 3].[355 5]>
/app/spider/hoon:<[343 3].[355 5]>
/app/spider/hoon:<[344 3].[355 5]>
/app/spider/hoon:<[347 5].[354 57]>
/app/spider/hoon:<[348 5].[354 57]>
/app/spider/hoon:<[349 7].[350 52]>
[%no-file-for-thread %azimuth]
/app/spider/hoon:<[350 7].
```

**Comment by wbsquared** [comment]
Actually the diagnose above was without the app being loaded. here it is after loading the app:

> -diagnose!azimuth
last downloaded %azimuth block: #15.373.409 on ~2022.8.19..20.28.21
last processed %azimuth block: #0 on unknown
installed %eth-watcher? %.y
running %eth-watcher? %.y
installed %azimuth? %.y
has %eth-watcher an %azimuth subscription? %.y
is %azimuth subscribed to %eth-watcher? %.y
...
(wait 5s. to run the thread again)



Note that I just now ran the series of commands to try to refresh the azimuth state (I have done this 4 times and still had been on block 15.373.409. We shall see..

**Comment by zalberico** [comment]
```
last processed %azimuth block: #0 on unknown
```
That sticks out as odd, @belisarius222, @yosoyubik, or @joemfb any idea what's going on here?

@wbsquared has a ship stuck and he's accessible to help us debug with it.

This is the most insidious bug I see people on the network have to deal with regularly.

**Comment by yosoyubik** [comment]
@zalberico  `#0 on unknown` means that after running `-azimuth-load` all the logs in /app/azimuth are deleted. next steps is to wait for %eth-watcher to download new ones (`last downloaded %azimuth block`), that are then sent to /app/azimuth to be processed.

The output from @wbsquared doesn't show the same issue I described here https://github.com/urbit/urbit/issues/5748#issuecomment-1374428270—both azimuth and eth-watcher have their subscriptions in place.

There could be something related to %behn timers—I'll update the diagnostic tool to look into it. Another option would be to use a (free) Infura node here `:azimuth|watch 'http://eth-mainnet.urbit.org:8545' %default` instead of the one provided by default.

**Comment by wbsquared** [comment]
I fixed ~witbep-witbep or the ship fixed itself. Here is what I did:

|pack
|meld

Shutdown with Ctrl-D, restart with --loom 32 (again - it already had been started with a 4 GB loom)

|nuke %astrolabe (I have no idea if this did anything but it was mentioned above so I did it even though I do like and use this app)

|nuke %eth-watcher
|start %eth-watcher
:azimuth %resub
:azimuth|watch 'http://eth-mainnet.urbit.org:8545' %default
-azimuth-load

Wait a night.

I am at 16.484.078 now.

**Comment by zalberico** [comment]
Nice @wbsquared! Glad you're unblocked and thanks for updating us with details. I remember people having issues with astrolabe and eth watcher before so it's definitely possible there's something there.

@yosoyubik thanks - this diagnostic tool and knowing how to change to a different node are really good tools to have in the troubleshooting toolkit. I'll post stuff here as I find people with the issue on the network.

**Comment by wbsquared** [comment]
yeah. note on my results from the diagnostic app that I had this:

has %eth-watcher an %azimuth subscription? %.y

whereas @yosoyubik had a negative result there.

**Comment by zalberico** [comment]
Yeah I saw that, just made your case weirder.

I suspect there are multiple causes of this bug, one of which may be related to astrolabe.

**Comment by thelifeandtimes** [comment]
I haven't had astrolabe installed for quite some time, but IIRC it does interact a decent amount with %azimuth or %eth-watcher and it put a very large obligation on my loom back before we had the bigger loom. Might be worth connecting w/ ~midlev-mindyr on the specific interaction mechanism?

**Comment by zalberico** [comment]
We debugged an example of this on @lukebuehler's ship with @philipcmonk and in his case when we went to the /~debug app under behn and searched for eth there was no timer despite the subscriptions and everything being set up correctly, eth watcher thought it was running and had a timer (but it did not).

In order to fix this, @philipcmonk made a code change in app/ethwatcher and will add the specifics here.

**Comment by philipcmonk** [comment]
We made wait-shortcut run unconditionally here:: https://github.com/urbit/urbit/blob/463a94101a854d9eec75b574499b78329af72d1c/pkg/arvo/app/eth-watcher.hoon#L351-L353

Then, you can run `:azimuth|listen ~ %app %azimuth` to force-restart the timer.  If you do this repeatedly, you might get multiple timers, which is not ideal but not the end of the world.

**Comment by zalberico** [comment]
@yosoyubik it'd be good to add a check for the timer to the diagnose tool if possible

**Comment by rabsef-bicrym** [comment]
I've recently booted a planet on it's first life, 0th rift.

![image](https://user-images.githubusercontent.com/52549148/219584027-fedaa533-4132-4d4f-90f3-f5331620c5c7.png)

It's so new, I still have the first event in my terminal scrollback. I have, to date, installed one application that I'm building and gotten SSL through acme. That's about it.

Now, I've started a flow I've seen on several other ships that isn't listed here as far as I can tell:
```hoon
gall: not running %vapor-ware yet, got %raw-poke
gall: not running %vapor-ware yet, got %raw-poke
gall: not running %vapor-ware yet, got %raw-poke
gall: not running %vapor-ware yet, got %raw-poke
ames: heard bad crypto (300 total), check azimuth state
ames: heard bad crypto (400 total), check azimuth state
ames: heard bad crypto (500 total), check azimuth state
ames: heard bad crypto (600 total), check azimuth state
ames: heard bad crypto (700 total), check azimuth state
ames: heard bad crypto (800 total), check azimuth state
ames: heard bad crypto (900 total), check azimuth state
> +azimuth-block
16.647.032
```

At the time of this writing, that block is roughly up to date. This heard bad crypto will grow from here. I've also seen it on other ships where the azimuth-block is behind. It doesn't seeeem to interfere with comms that already existed for ~fipfun-podped but I'm not really doing anything on that ship but testing software that requires an SSL connection, so, I don't really know if it can't communicate with people.

On ~hiddev-dannut, I'm on my 3rd/4th day of hoping that various attempts to get the block to update will work. She's currently in local only, hopefully still getting eth information and waiting to update. Otherwise she just gets number go up here: `ames: packet dropped (577000 total)`.

I'm not sure if these things relate but I will say that something recently caused a lot of instability across ships that I maintain and I'm chasing the fires w/ little success.

**Comment by rabsef-bicrym** [comment]
An aside but "check azimuth state" suggests to me to IDK :azimuth +dbug which is a bad idea or maybe -load-azimuth which will put me back and leave me hoping that I'll catch up, so I'm a little stultified for options.

**Comment by yosoyubik** [comment]
@rabsef-bicrym have you checked if /app/azimuth has received any logs? (check with `:azimuth +dbug state+'(lent logs)'` if you don't have `-diagnose!azimuth` installed. I found an [instance](https://github.com/urbit/urbit/issues/5748#issuecomment-1374428270) where azimuth and eth-watcher were not in sync, so +azimuth-block reported an up-to-date block, but in reality /app/azimuth had not processed anything (and then those bad crypto messages would make sense since you have out of date keys for people that have since rotated them)

**Comment by rabsef-bicrym** [comment]
@yosoyubik 

I had previously done diagnose-azimuth on all my ships w/ this problem and they were all listing as "%.y" and other positive outcomes for the test. I had not done logs.

Hiddev, after sitting all night in offline mode, finally caught a block and processed some breaches. It is now up to date and appears to be actually, functionally online (yet to see an Ames packet drop).

fipfun-podped does appear to continue to have issues:

After downloading diagnose, fipfun produced the following results for azimuth:
```hoon
> -diagnose!azimuth
last downloaded %azimuth block: #16.649.636 on ~2023.2.17..16.33.59
last processed %azimuth block: #16.649.174 on ~2023.2.17..15.00.11
installed %eth-watcher? %.y
running %eth-watcher? %.y
installed %azimuth? %.y
has %eth-watcher an %azimuth subscription? %.y
has %eth-watcher a %behn timer? %.y
is %azimuth subscribed to %eth-watcher? %.y
...
(wait 5s. to run the thread again)
```
Log lent is 13.918.

Number has continued to go up, nonetheless:
```
ames: heard bad crypto (1400 total), check azimuth state
ames: heard bad crypto (1500 total), check azimuth state
```
Since fipfun has no friends, this number go up slowly.

**Comment by rabsef-bicrym** [comment]
I spoke too soon - hiddev getting some packet drops but maybe fewer?

**Comment by pkova** [comment]
We have not seen this in many many years.
